modules/websocket/websocket-source.c

```diff
@@ -0,0 +1,212 @@
+  WebsocketSrcDriver *self = (WebsocketSrcDriver *)source;
+  g_free(self->protocol);
+  self->protocol = g_strdup(protocol);
+}
+
+void
+websocket_sd_set_path(LogDriver *source, gchar *path) {
+  WebsocketSrcDriver *self = (WebsocketSrcDriver *)source;
+  g_free(self->path);
+  self->path = g_strdup(path);
+}
+
+void websocket_sd_set_enable_ssl(LogDriver *source, int flag) {
+  WebsocketSrcDriver *self = (WebsocketSrcDriver *)source;
+  self->enable_ssl = flag;
+}
+
+void websocket_sd_set_cert(LogDriver *source, gchar *path) {
+  WebsocketSrcDriver *self = (WebsocketSrcDriver *)source;
+  g_free(self->cert);
+  self->cert = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_sd_set_key(LogDriver *source, gchar *path) {
+  WebsocketSrcDriver *self = (WebsocketSrcDriver *)source;
+  g_free(self->key);
+  self->key = path == NULL ? NULL : g_strdup(path);
+}
+
+void websocket_sd_set_cacert(LogDriver *source, gchar *path) {
+  WebsocketSrcDriver *self = (WebsocketSrcDriver *)source;
+  g_free(self->cacert);
+  self->cacert = path == NULL ? NULL : g_strdup(path);
+}
+
+
+static inline gboolean
+create_reader(LogPipe *s)
+{
+  int ret;
+  int fd;
+  WebsocketSrcDriver* self = (WebsocketSrcDriver *)s;
+  GlobalConfig *cfg = log_pipe_get_config(s);
+
+  if (websocket_server_create(self->protocol, self->port, self->enable_ssl, self->cert, self->key, self->cacert, &fd, &self->msgqid, &self->service_pid))
+    return FALSE;
+  LogTransport* transport = log_transport_websocket_new(fd);
+  PollEvents* poll_events = poll_fd_events_new(transport->fd);
+  LogProtoServerOptions* proto_options = &self->reader_options.proto_options.super;
+  LogProtoServer* proto = log_proto_text_server_new(transport, proto_options);
+
+  self->reader = log_reader_new(cfg);
+
+  self->reader_options.parse_options.flags |= LP_NOPARSE;
+
+  log_reader_reopen(self->reader, proto, poll_events);
+  return TRUE;
+}
+
+static gboolean
+websocket_sd_init(LogPipe *s)
+{
+  WebsocketSrcDriver *self = (WebsocketSrcDriver *) s;
+  GlobalConfig* cfg = log_pipe_get_config(&self->super.super.super);
+
+  if (!log_src_driver_init_method(s))
+  {
+    msg_error("Failed to initialize source driver!", NULL);
+    return FALSE;
+  }
+
+  log_reader_options_init(&self->reader_options, cfg, "websocket");
+
+  if (!create_reader(s))
+    return FALSE;
+
+  log_reader_set_options(self->reader,
+                             s,
+                             &self->reader_options,
+                             STATS_LEVEL1,
+                             SCS_WEBSOCKET,
+                             self->super.super.id,
+                             "websocket");
+  log_pipe_append((LogPipe *) self->reader, s);
+
+  if (!log_pipe_init((LogPipe *) self->reader))
+  {
+    msg_error("Error initializing log_reader", NULL);
+    log_pipe_unref((LogPipe *) self->reader);
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+
+static gboolean
+websocket_sd_deinit(LogPipe *s)
```

Comment:
I tested reload syslog-ng with `kill -1 PID` when reloading syslog-ng. The websocket module will connect automatically with the new script when reloaded.
On the other hand, libwebsocket didn't provide an api such as `zmq_ctx_new` or `zmq_socket`. 
Reconnecting with the new configuration may be a better way to survive from reloading.